Add edge prediction subsection and supplement (#11)
greenelab/xswap-manuscript: content/02.body.md

## Discussion

### Prediction performance and the edge prior

> We focus on (biomedical) network edge prediction.

Reviewer: ```suggestion
We focus on edge prediction in biomedical networks.
```

> In particular, we are interested in predicting new edges with specificity, so that predictions reflect particular connectivity rather than generic node characteristics.

Reviewer: ```suggestion
Our overall goal is to predict new edges with specificity, so that predictions reflect particular connectivity rather than generic node characteristics.
```

> We believe that degree is a broadly appropriate proxy for nonspecificity, meaning that predictive performance attributable to degree can be considered nonspecific across a variety of domains.

Reviewer: ```suggestion
Our software is designed to capture predictive performance attributable to degree to provide a baseline expectation for edge pairs.
We expect that degree-based non-specificity is not a unique property of biomedical networks
```

For example, if node A connects to nearly all other nodes in a network, predicting that all remaining nodes share an edge with node A will likely result in many correct--though nonspecific--predictions, regardless of the type of data contained in the network.

> Node degree must be accounted for, rather than eliminated, to make correct predictions while being able to distinguish specific from nonspecific predictions.

Reviewer: ```suggestion
Node degree should be accounted for to make correct predictions while being able to distinguish specific from nonspecific predictions.
```

Prediction without reliance on node degree is challenging because many effective methods for edge prediction are correlated with degree (Figure {@fig:feature-degree}).

> Therefore, effective predictive methods will make both specific and nonspecific predictions.

Reviewer: ```suggestion
```

The effects of node degree are obvious when edge prediction features are functions of degree.
For example, the resource allocation index is the sum of inverse degree of common neighbors between source and target nodes (in the symmetric case), while preferential attachment is the product of source and target degree [@doi:10.1140/epjb/e2009-00335-8; @doi:10.1145/1065385.1065415].
However, because many other edge prediction methods are not explicitly degree-based, it is important to have a general method for comparing the effects of node degree on edge prediction methods.

> To account for degree without impairing predictive performance, we are proposing the quantification of edge probability due to degree.

Reviewer: ```suggestion
We develop software that quantifies the edge probability due to degree.
```

We term this probability the "edge prior", and we have identified two applications.
First, a probability associated with every node pair can be treated as a classification score.

> Ordering these scores allows the assessment of diagnostic ability (eg: via AUROC) and calibration.

Reviewer: ```suggestion
Ordering these scores provides an assessment of performance based solely on degrees and edge distributions, which can be used as a baseline for other classifiers.
```

> The edge prior can be used to find a degree-based performance baseline when comparing and evaluating other classifiers.

Reviewer: ```suggestion
```
